semgrep-core/src/core/il/IL_helpers.ml

```diff
@@ -144,6 +146,30 @@ let rlvals_of_node = function
   | NTodo _ ->
       []
 
+(* Helpers for specific nodes *)
+
+let call_assign_of_string = function
+  | "add" -> Add
+  | "addAll" -> AddAll
+  | _else_ -> failwith "Unexpected call_assign"
+
+let should_clean_lval x =
+  match x.i with
+  | Assign _
+  | AssignAnon _
+  | Call _
+  | CallSpecial _
+  | FixmeInstr _ ->
+      true
+  | CallAssign (_, ((Add | AddAll), _), _) ->
```

Comment:
Do you have bugs in the test suite that need this? If you do `x[i] = clean` we assume that `x` gets cleaned and, although I'm aware that it's not sound, it probably avoids FPs. Rather than making this ad-hoc for Java collections, I would prefer that we have a rule option that let's you choose what kind of assumptions to make. I just recently added `taint_assume_safe_functions` and `taint_assume_safe_indexes` in that spirit. (I know options aren't great but...). Then DeepSemgrep Java could start having different defaults than Semgrep.